lib/assets/assets.go

```diff
@@ -53,19 +53,28 @@ func Serve(w http.ResponseWriter, r *http.Request, asset Asset) {
 		return
 	}
 
-	if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+	switch {
+	case !gzipped(asset.Content):
+		header.Set("Content-Length", strconv.Itoa(len(asset.Content)))
+		io.WriteString(w, asset.Content)
+	case strings.Contains(r.Header.Get("Accept-Encoding"), "gzip"):
 		header.Set("Content-Encoding", "gzip")
-		header.Set("Content-Length", strconv.Itoa(len(asset.ContentGz)))
-		io.WriteString(w, asset.ContentGz)
-	} else {
+		header.Set("Content-Length", strconv.Itoa(len(asset.Content)))
+		io.WriteString(w, asset.Content)
+	default:
 		// gunzip for browsers that don't want gzip.
 		var gr *gzip.Reader
-		gr, _ = gzip.NewReader(strings.NewReader(asset.ContentGz))
+		gr, _ = gzip.NewReader(strings.NewReader(asset.Content))
 		io.Copy(w, gr)
 		gr.Close()
 	}
 }
 
+func gzipped(content string) bool {
+	const gzmagic = "\x1f\x8b" // RFC 1952, §2.3.1.
```

Comment:
We should just have a flag/enum for the compression used, if any.

Reply:
At which point we can also add an `length int` and serve that as `Content-Length` to clients not speaking gzip.

Reply:
That would be nicer, but the code for it is much more complicated. The generated code and the client in lib/assets need some common type to communicate bool+string.

Reply:
The `Asset` struct?

Reply:
Yes. Turned out to be easier than I thought, see updated code.